src/helics/core/mpi/MpiBroker.cpp

```diff
@@ -10,64 +10,67 @@ Lawrence Livermore National Laboratory, operated by Lawrence Livermore National
 */
 #include "MpiBroker.h"
 #include "MpiComms.h"
-#include "helics/common/blocking_queue.h"
-#include "helics/config.h"
-#include "helics/core/core-data.h"
-#include "helics/core/core.h"
-#include "helics/core/helics-time.h"
 
-#include <algorithm>
-#include <cassert>
-#include <chrono>
-#include <cstdint>
-#include <cstring>
-#include <fstream>
-#include <sstream>
+#include "../../common/argParser.h"
 
-#include "helics/core/argParser.h"
-
-static const std::string DEFAULT_BROKER = "tcp://localhost:5555";
+#include <mpi.h>
 
 namespace helics
 {
-using namespace std::string_literals;
-static const argDescriptors extraArgs{{"brokerinit"s, "string"s, "the initialization string for the broker"s}};
+    MpiBroker::MpiBroker(bool rootBroker) noexcept : CommsBroker(rootBroker) {}
 
-MpiBroker::MpiBroker (bool rootBroker) noexcept : CommsBroker (rootBroker) {}
+    MpiBroker::MpiBroker(const std::string &broker_name) : CommsBroker(broker_name) {}
 
-MpiBroker::MpiBroker (const std::string &broker_name) : CommsBroker (broker_name) {}
+    MpiBroker::~MpiBroker() = default;
 
-MpiBroker::~MpiBroker () {}
+    using namespace std::string_literals;
+    static const ArgDescriptors extraArgs{ { "broker_address", ArgDescriptor::arg_type_t::string_type, "location of a broker using mpi (rank:tag)" } };
```

Comment:
I might suggest having the option of this being two separate arguments as well.  Then tag can reuse the name field if need.  Not a required change but as we evaluate and adapt this might be worth a small change to better match the other brokers

Reply:
I think it would be a --brokername field as well.  In any case there is a name associated with a broker/core  either specified or autogenerated from a guid

Reply:
Okay. I'm not sure how well reusing it for the tag would work, since the tag has to be an integer. Maybe a hash of the name or autogenerated guid would work.

Reply:
didn't catch that it had to be an integer for the tag, thought it was a string,  in that case ignore my comment other than you should be able to specify the rank and tag separately